packages/@ourworldindata/grapher/src/controls/Controls.tsx

```diff
@@ -1,18 +1,500 @@
+                    onClick={(): void => {
+                        this.props.manager.isSelectingData = !active
+                    }}
+                >
+                    <FontAwesomeIcon icon={icon} /> {label}
+                </button>
+            </div>
+        ) : null
+    }
+}
+
+export interface SettingsMenuManager {
+    showConfigurationDrawer?: boolean
+
+    // ArchieML directives
+    hideFacetControl?: boolean
+    hideRelativeToggle?: boolean
+    hideEntityControls?: boolean
+    hideZoomToggle?: boolean
+    hideNoDataAreaToggle?: boolean
+    hideFacetYDomainToggle?: boolean
+    hideXScaleToggle?: boolean
+    hideYScaleToggle?: boolean
+
+    // chart state
+    type: ChartTypeName
+    isRelativeMode?: boolean
+    selection?: SelectionArray | EntityName[]
+    filledDimensions: ChartDimension[]
+    xColumnSlug?: string
+    xOverrideTime?: number
+    hasTimeline?: boolean
+    canToggleRelativeMode: boolean
+
+    // linear/log & align-faceted-axes
+    yAxis: AxisConfig
+    xAxis: AxisConfig
+
+    // zoom-to-selection
+    zoomToSelection?: boolean
+
+    // show no-data entities in marimekko
+    showNoDataArea?: boolean
+
+    // facet by
+    availableFacetStrategies: FacetStrategy[]
+    facetStrategy?: FacetStrategy
+    entityType?: string
+    facettingLabelByYVariables?: string
+
+    // absolute/relative units
+    stackMode?: StackMode
+    relativeToggleLabel?: string
+
+    // show intermediate scatterplot points
+    compareEndPointsOnly?: boolean
+}
+
+@observer
+export class SettingsMenu extends React.Component<{
+    manager: SettingsMenuManager
+    top: number
+    bottom: number
+}> {
+    @observable.ref visible: boolean = false
+    @observable.ref shouldRender: boolean = false
+
+    @computed get showYScaleToggle(): boolean | undefined {
+        if (this.manager.hideYScaleToggle) return false
+        if (this.manager.isRelativeMode) return false
+        if ([StackedArea, StackedBar].includes(this.manager.type)) return false // We currently do not have these charts with log scale
+        return this.manager.yAxis.canChangeScaleType
+    }
+
+    @computed get showXScaleToggle(): boolean | undefined {
+        if (this.manager.hideXScaleToggle) return false
+        if (this.manager.isRelativeMode) return false
+        return this.manager.xAxis.canChangeScaleType
+    }
+
+    @computed get showFacetYDomainToggle(): boolean {
+        // don't offer to make the y range relative if the range is discrete
+        return (
+            !this.manager.hideFacetYDomainToggle &&
+            this.manager.facetStrategy !== FacetStrategy.none &&
+            this.manager.type !== StackedDiscreteBar
+        )
+    }
+
+    @computed get showZoomToggle(): boolean {
+        // TODO:
+        // grapher passes a SelectionArray instance but programmatically defined
+        // managers treat `selection` as a string[] of entity names. do we need both?
```

Comment:
Not sure why selection is sometimes treated as SelectionArray instance, sometimes as string[]. That's probably a question for @marcelgerber.

I've seen the [makeSelectionArray](https://github.com/owid/owid-grapher/blob/a3d32f96431cbc43b74d8d00b9539432c52044d5/packages/%40ourworldindata/grapher/src/chart/ChartUtils.tsx#L80) in use, so you could use that here as well?